In [14]:
import numpy as np
import pandas as pd
import scipy.stats as stats
import matplotlib.pyplot as plt
import seaborn as sns

%matplotlib inline

### It is assumed that the mean systolic blood pressure is μ = 120 mm Hg. In the Honolulu Heart Study, a sample of n = 100 people had an average systolic blood pressure of 130.1 mm Hg with a standard deviation of 21.21 mm Hg. 

### Is the group significantly different (with respect to systolic blood pressure!) from the regular population?

In [ ]:
Yes. 

### Set up the hypothesis test.

In [ ]:
H0 = μ = 120 
Ha = μ != 120 

n = 100 
sample_mean = 130.1
standard_deviation = 21.21

In [8]:
t = (130.1 - 120)/(21.21/10)
t

4.761904761904759

In [15]:
tc =  1.660 

In [ ]:
t > tc ---- do not reject null hypothesis

### Calculate the test statistic by hand and also code it in Python. It should be 4.76190. What decision can you make based on this calculated value?

In [18]:
H0 = μ = 120 
Ha = μ != 120 

n = 100 
sample_mean = 130.1
pop_mean = 120
standard_deviation = 21.21

In [22]:
t = (sample_mean - pop_mean)/ (standard_deviation/ np.sqrt(100))
print("Our t value is: {:.2f}".format(t))

Our t value is: 4.76


In [23]:
if t < tc :
    print("Reject H0")
else: 
    print("Do not reject H0")

Do not reject H0


### Optional : 
In a packing plant, a machine packs cartons with jars. It is supposed that a new machine will pack faster on the average than the machine currently used. To test that hypothesis, the times it takes each machine to pack ten cartons are recorded. The results, in seconds, are shown in the tables in the file Data/machine.txt. Assume that there is sufficient evidence to conduct the t test, does the data provide sufficient evidence to show if one machine is better than the other.
Hint: use two sample t-test

t= (sample_mean(x1) - sample_mean(x2)) / sqrt(square(s1)/n1 + square(s2)/n2 )

New Machine	  Old Machine
42.1	        42.7
41	          43.6
41.3	        43.8
41.8	        43.3
42.4	        42.5
42.8	        43.5
43.2	        43.1
42.3	        41.7
41.8	        44
42.7	        44.1

H0: μ1 ≠ μ2
Ha: μ1 = μ2

In [26]:
New_machine_mean = ((42.1 + 41 + 41.3 + 41.8 + 42.4 + 42.8 + 43.2 +42.3 + 41.8 +42.7)/10)
New_machine_mean

42.14

In [28]:
Old_machine_mean = ((42.7 + 43.6 + 43.8 + 43.3 + 42.5 + 43.5 + 43.1 + 41.7 +44 + 44.1)/10)
Old_machine_mean

43.230000000000004

In [30]:
S1_square = 0.4671
S1_square

0.4671

In [31]:
S2_square = 0.5623
S2_square

0.5623

In [40]:
t = (New_machine_mean - Old_machine_mean) / ((np.sqrt(S1_square))/10 + (np.sqrt(S2_square))/10)
t

-7.604754800709014